test_conformance/atomics/test_atomics.cpp

```diff
@@ -200,6 +200,10 @@ int test_atomic_function(cl_device_id deviceID, cl_context context, cl_command_q
         error = clGetKernelWorkGroupInfo( kernel, deviceID, CL_KERNEL_WORK_GROUP_SIZE, sizeof( workSize ), &workSize, NULL );
         test_error( error, "Unable to obtain max work group size for device and kernel combo" );
 
+        // Limit workSize to avoid extremely large local buffer size and slow
+        // run.
+        if (testFns.NumResultsFn && workSize > 65536) workSize = 65536;
+
         // "workSize" is limited to that of the first dimension as only a 1DRange is executed.
         if( maxSizes[0] < workSize )
```

Comment:
What about limitation of `workSize` based on this condition. Watching Intel platforms this values in `maxSizes[0]` are much smaller than 65536. Could you deliver information what values you have in `maxSizes[0]` and `workSize` . I suppose you have broken values in `maxSizes` array.

Reply:
Additionally please check value `CL_DEVICE_MAX_WORK_GROUP_SIZE`. According to specification  https://www.khronos.org/registry/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_KERNEL_WORK_GROUP_SIZE
`CL_KERNEL_WORK_GROUP_SIZE will be less than or equal to CL_DEVICE_MAX_WORK_GROUP_SIZE for a given kernel object.`

Reply:
@gwawiork, thanks for comments!

With Intel OpenCL CPU runtime, the value 8192, while with Intel FPGA emulator, the max allowed work group size is `64 * 1024 * 1024`, which is quite large. And it can require huge amount of memory and can also be slow to run the test on FPGA emulator (which is actually a CPU device).

Reply:
I do not understand. Could you write once again what value you have in maxSizes[0] (CL_DEVICE_MAX_WORK_ITEM_SIZES) and in workSize (CL_KERNEL_WORK_GROUP_SIZE) ?

Reply:
`maxSizes[0] = workSize = 64 * 1024 * 1024 = 67108864`

Reply:
Form me this code is ok. 
Just additional suggestion/idea.  It would be useful to add user command parameter to reduce or force given value just make these test runnable on emulation/silicon environment in different conditions.